src/Service/FilterFactory.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Service;
+
+use Lmc\ApiFilter\Constant\Filter;
+use Lmc\ApiFilter\Entity\Value;
+use Lmc\ApiFilter\Filter\FilterIn;
+use Lmc\ApiFilter\Filter\FilterInterface;
+use Lmc\ApiFilter\Filter\FilterWithOperator;
+
+class FilterFactory
+{
+    public function createFilter(string $column, string $filter, Value $value, string $title = null): FilterInterface
+    {
+        switch (mb_strtolower($filter)) {
+            case Filter::EQUALS:
+                return new FilterWithOperator($column, $value, '=', $title ?? Filter::EQUALS);
+            case Filter::GREATER_THAN:
+                return new FilterWithOperator($column, $value, '>', $title ?? Filter::GREATER_THAN);
+            case Filter::LESS_THEN:
+                return new FilterWithOperator($column, $value, '<', $title ?? Filter::LESS_THEN);
+            case Filter::LESS_THEN_OR_EQUAL:
+                return new FilterWithOperator($column, $value, '<=', $title ?? Filter::LESS_THEN_OR_EQUAL);
+            case Filter::GREATER_THAN_OR_EQUAL:
+                return new FilterWithOperator($column, $value, '>=', $title ?? Filter::GREATER_THAN_OR_EQUAL);
+            case Filter::IN:
+                return new FilterIn($column, $value, $title);
+        }
+
+        throw new \InvalidArgumentException(
```

Comment:
It would be lovely (one more reason to get https://github.com/lmc-eu/api-filter/pull/26 done :) ) to have something like:
 
```suggestion
        throw InvalidFilterException::forNotImplementedFilter($filter, $column, $value);
```

And move the logic there. Those custom exceptions with static constructors helps greatly to cleanup the overhead and string mambo jumbo when throwing exceptions :).

Reply:
I will do this after PR #26 is merged.